src/components/map/MapService.js

```diff
@@ -240,6 +244,10 @@ goog.require('ga_urlutils_service');
           updateDelay: {
             writable: true,
             value: null
+          },
+          userVisible: {
+            writable: true,
+            value: olLayer.getVisible()
```

Comment:
It's happening only for layer added asynchronously (kml file).

For these layers we pass the visible property directly in the constructor before the call to `gaDefineLayerProperties` so the overrided `setVisible` function is never called so `userVisible` is always set to true even if the visibility permalink says it should be false.